packages/iTwinUI-react/src/core/TimePicker/TimePicker.tsx

```diff
@@ -470,4 +653,108 @@ const TimePickerColumn = <T,>(props: TimePickerColumnProps<T>): JSX.Element => {
   );
 };
 
+type VirtualizedTimePickerColumnProps<T = Date> = {
+  /**
+   * What value to display in every cell.
+   */
+  valueRenderer: (value: T, precision: Precision) => React.ReactNode;
+  /**
+   * Precision of the time.
+   * @default 'minutes'
+   */
+  precision: Precision;
+} & ClassNameProps &
+  Omit<TimePickerColumnProps<T>, 'valueRenderer'>;
+
+// const VirtualizedTimePickerColumn = <T,>(
+//   props: VirtualizedTimePickerColumnProps<T>,
+// ): JSX.Element => {
+
+const VirtualizedTimePickerColumn = <T,>(
+  props: VirtualizedTimePickerColumnProps<T>,
+): JSX.Element => {
+  const {
+    data,
+    onFocusChange,
+    onSelectChange,
+    isSameFocused,
+    isSameSelected,
+    setFocus = false,
+    valueRenderer,
+    precision = 'minutes',
+    className = 'iui-time',
+  } = props;
+
+  const needFocus = React.useRef(setFocus);
+
+  // Used to focus row only when changed (keyboard navigation)
+  // e.g. without this on every rerender it would be focused
+  React.useEffect(() => {
+    if (needFocus.current) {
+      needFocus.current = false;
+    }
+  });
+
+  const virtualizedItemRenderer = React.useCallback(
+    (index: number) => {
+      const value = data[index];
+      const isSameFocus = isSameFocused(value);
+      return (
+        <li
+          onKeyDown={(event) => {
+            handleTimeKeyDown(
+              event,
+              data.length - 1,
+              (index) => onFocusChange(data[index]),
+              (index) => onSelectChange(data[index]),
+              index,
+              needFocus,
+            );
+          }}
+          className={cx({
+            'iui-selected': isSameSelected(value),
+          })}
+          key={index}
+          tabIndex={isSameFocus ? 0 : undefined}
+          onClick={() => {
+            onSelectChange(value);
+          }}
+        >
+          {valueRenderer(value, precision)}
+        </li>
+      );
+    },
+    [
+      data,
+      isSameFocused,
+      isSameSelected,
+      onFocusChange,
+      onSelectChange,
+      precision,
+      valueRenderer,
+    ],
+  );
+
+  const currentDateIndex = () => {
+    for (let i = 0; i < data.length; i++) {
+      if (isSameSelected(data[i])) {
+        return i;
+      }
+    }
+    return -1;
+  };
+
+  const { visibleChildren } = useVirtualization({
```

Comment:
useVirtualization returns `{ outerProps, innerProps, visibleChildren }`. I think we also might have to apply `outerProps` and `innerProps` as in the JSDocs example.

https://github.com/iTwin/iTwinUI-react/blob/21900d42900f7a03305bf2040cc64ddb29361b2d/packages/iTwinUI-react/src/core/utils/components/VirtualScroll.tsx#L146-L152

Reply:
I removed one of them because it was negatively affecting the css. I think I removed the other one in the course of trying to figure out some bugs. But I will reinstate them and try that